core/src/apps/cardano/sign_tx.py

```diff
@@ -146,18 +116,24 @@ def _process_outputs(self):
         change_coins = []
 
         for output in self.outputs:
-            if output.address_n:
-                address, _ = derive_address_and_node(self.keychain, output.address_n)
+            if output.address_parameters:
+                address = derive_address(
+                    self.keychain,
+                    output.address_parameters,
+                    self.protocol_magic,
+                    human_readable=False,
+                )
                 change_addresses.append(address)
-                change_derivation_paths.append(output.address_n)
+                change_derivation_paths.append(output.address_parameters.address_n)
```

Comment:
That `address_n` field irritates me quite a lot, perhaps we should rename it to something meaningful. Not in this PR, though.

Reply:
Ok, I agree, added to my todo list.